### PR TITLE
add ripgrep to travis ci cache directories list

### PR DESCRIPTION
**Pull request by tom-shan**
CI often fails for github api rate limit when downloading
ripgrep package. This workaroud is trying to enable
Travis CI cache of ripgrep package.

### CODE REVIEW

**Comment by kittaakos** [comment]
Hopefully, this PR fixes https://github.com/theia-ide/theia/issues/3787.

File: .travis.yml [hunk 1]
```diff
@@ -66,6 +66,7 @@ cache:
 # end_cache_directories
   - packages/java-debug/download
   - packages/debug-nodejs/download
+  - /tmp/vscode-ripgrep-cache-1.2.4
```
Comment (akosyakov) [inline .travis.yml]: only concern that we should keep it sync with rigrep version, maybe we can improve `scripts/prepare-travis`

Comment (tom-shan) [reply .travis.yml]: I will have a try

Comment (tom-shan) [reply .travis.yml]: done

**Comment by marcdumais-work** [comment]
Assuming this works as intended, would there be a way to re-use this strategy when building the docker images in [theia-apps](https://github.com/theia-ide/theia-apps)?

**Comment by akosyakov** [comment]
@marcdumais-work yes, merging, if it helps then we can try with theia-apps

**Comment by tom-shan** [comment]
I notice that a couple of recent CI results are still failed. During these builds, there are some weird logs about Travis cache, for example https://travis-ci.org/theia-ide/theia/jobs/547289068#L248 and https://travis-ci.org/theia-ide/theia/jobs/547291207#L248.

**Comment by marcdumais-work** [comment]
@tom-shan thanks for pointing this out. 

Do we know if caching ripgrep, as done in this PR, has been observed to work for PRs originating from a non-committer's fork? 

I'll clear the current Travis cache for our project, in case it might help.

**Comment by tom-shan** [comment]
Yes, you can refer to https://docs.travis-ci.com/user/caching/#pull-request-builds-and-caches

I'm not sure whether clearing cache really help. But after you clear the master cache, you should recreate it as soon as possible, or the new PR is going to download the ripgrep from github and may fail the CI.

**Comment by marcdumais-work** [comment]
>  it should be pointed out that after the master cache is cleared, you'd better recreate it as soon as possible

+1. Not practically a problem however; the cache was recreated minutes after I cleared it - @akosyakov had a PR being tested that repopulated the cache.

> Yes, you can refer to https://docs.travis-ci.com/user/caching/#pull-request-builds-and-caches

Thanks - that does not prove that the mechanism works as we intended - so far I have not seen a Travis run that succeeds for a non-committer.

**Pull request merged**